netbox/utilities/testing/api.py

```diff
@@ -447,34 +450,34 @@ def _build_query(self, name, **filters):
 
             # Compile list of fields to include
             fields_string = ''
-            for field_name, field in type_class._meta.fields.items():
-                is_string_array = False
-                if type(field.type) is GQLList:
-                    if field.type.of_type is GQLString:
-                        is_string_array = True
-                    elif type(field.type.of_type) is GQLNonNull and field.type.of_type.of_type is GQLString:
-                        is_string_array = True
-
-                if type(field) is GQLDynamic:
-                    # Dynamic fields must specify a subselection
-                    fields_string += f'{field_name} {{ id }}\n'
-                # TODO: Improve field detection logic to avoid nested ArrayFields
-                elif field_name == 'extra_choices':
-                    continue
-                elif inspect.isclass(field.type) and issubclass(field.type, GQLUnion):
-                    # Union types dont' have an id or consistent values
+
+            for field in type_class.__strawberry_definition__.fields:
+                if (
+                    field.type in (
+                        strawberry_django.fields.types.DjangoFileType, strawberry_django.fields.types.DjangoImageType) or
+                    type(field.type) is StrawberryOptional and field.type.of_type in (
+                        strawberry_django.fields.types.DjangoFileType, strawberry_django.fields.types.DjangoImageType)
+                ):
```

Comment:
We can clean this up by defining a list of fields to check for.
```suggestion
                file_fields = (strawberry_django.fields.types.DjangoFileType, strawberry_django.fields.types.DjangoImageType)
                if (
                    field.type in file_fields or (
                        type(field.type) is StrawberryOptional and field.type.of_type in file_fields
                    )
                ):
```